addon-resizer/nanny/main/pod_nanny.go

```diff
@@ -233,3 +237,99 @@ func decodeNannyConfiguration(data []byte, codecs *serializer.CodecFactory) (*na
 	}
 	return externalHC, nil
 }
+
+func updateResources(nannycfg *nannyconfig.NannyConfiguration, baseStorage string) []nanny.Resource {
+	var resources []nanny.Resource
+
+	// Monitor only the resources specified.
+	if nannycfg.BaseCPU != nannyconfig.NoValue {
+		resources = append(resources, nanny.Resource{
+			Base:         resource.MustParse(nannycfg.BaseCPU),
+			ExtraPerNode: resource.MustParse(nannycfg.CPUPerNode),
+			Name:         "cpu",
+		})
+	}
+
+	if nannycfg.BaseMemory != nannyconfig.NoValue {
+		resources = append(resources, nanny.Resource{
+			Base:         resource.MustParse(nannycfg.BaseMemory),
+			ExtraPerNode: resource.MustParse(nannycfg.MemoryPerNode),
+			Name:         "memory",
+		})
+	}
+
+	if baseStorage != nannyconfig.NoValue {
+		resources = append(resources, nanny.Resource{
+			Base:         resource.MustParse(baseStorage),
+			ExtraPerNode: resource.MustParse(nannycfg.MemoryPerNode),
+			Name:         "storage",
+		})
+	}
+
+	glog.Infof("Resources: %+v", resources)
+	return resources
+}
+
+type nannyConfigUpdater struct {
```

Comment:
Good point. I was thinking of adding unit tests for this new struct, but didn't do that in the first place because the only logic that's worth tests is CurrentResources function, but (1) it's pretty much the same as original loadNannyConfiguration func which doesn't have unit tests either, (2) the only difference is it gets configmap through api calls while in loadNannyConfiguration func it reads the file. And we cannot issue real api calls in unit tests. The only way to test is to make a fake one, which doesn't seem to be worthwhile?

But I do agree it's good to consider adding unit tests for the general loadNannyConfiguration logic. So I added a TODO here so we can improve later.

Reply:
I'd prefer adding the test in the same PR just from experience that "adding tests later" usually doesn't win in the priority game and then it never happens.

Reply:
Tests added. Run with `go test -v` and all passed.